python/ray/serve/_private/router.py

```diff
@@ -358,6 +386,39 @@ def target_num_scheduling_tasks(self) -> int:
     def curr_replicas(self) -> Dict[str, ReplicaWrapper]:
         return self._replicas
 
+    def _get_actor_name(self) -> str:
+        """Gets the name of the actor where this scheduler runs.
+
+        NOTE: this call hangs when the GCS is down. As long as this method is
+        called only when the scheduler is initialized, that should be
+        okay because a ServeHandle (and its scheduler) relies
+        on the Serve controller for intialization, and the Serve controller
+        is runs only when the GCS is up.
+
+        Return:
+            The name of the actor where this scheduler runs. If the scheduler
+            runs in the driver, returns "DRIVER". If the method fails, returns
+            an empty string.
+        """
+
+        if in_ray_driver_process():
+            return "DRIVER"
+        else:
+            try:
+                actor_id = ray.get_runtime_context().get_actor_id()
+                if actor_id is None:
+                    return ""
+                else:
+                    # NOTE (shrekris-anyscale): this import is delayed because
+                    # putting at the top of the file causes Ray client tests
+                    # to fail.
+                    from ray.util import state
```

Comment:
I needed to delay this import because it was causing Ray client tests to fail when it was at the top of the file. The tests that failed were:

* `test_quickstart_class`
* `test_quickstart_counter`
* `test_handle_hanging[serve_with_client0]`

For example, [`test_quickstart_class`](https://github.com/ray-project/ray/blob/b82b8643cc89dd87caf074d8e8bf342d0318fa74/python/ray/serve/tests/test_ray_client.py#L136) failed locally with following traceback:

```
Put failed:


―――――――――――――――――――――――――――― test_quickstart_class ―――――――――――――――――――――――――――――

serve_with_client = None

    def test_quickstart_class(serve_with_client):
>       serve.start()
    
        @serve.deployment

test_ray_client.py:137: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../api.py:107: in start
    client = _private_api.serve_start(detached, http_options, dedicated_cpu, **kwargs)
../_private/api.py:310: in serve_start
    controller, controller_name = _start_controller(
../_private/api.py:182: in _start_controller
    controller = ServeController.options(**controller_actor_options).remote(
../../actor.py:645: in remote
    return actor_cls._remote(args=args, kwargs=kwargs, **updated_options)
../../util/tracing/tracing_helper.py:385: in _invocation_actor_class_remote_span
    return method(self, args, kwargs, *_args, **_kwargs)
../../actor.py:769: in _remote
    return client_mode_convert_actor(self, args, kwargs, **actor_options)
../../_private/client_mode_hook.py:184: in client_mode_convert_actor
    return client_actor._remote(in_args, in_kwargs, **kwargs)
../../util/client/common.py:396: in _remote
    return self.options(**option_args).remote(*args, **kwargs)
../../util/client/common.py:602: in remote
    futures = ray.call_remote(self, *args, **kwargs)
../../util/client/api.py:100: in call_remote
    return self.worker.call_remote(instance, *args, **kwargs)
../../util/client/worker.py:556: in call_remote
    task = instance._prepare_client_task()
../../util/client/common.py:587: in _prepare_client_task
    task = self._remote_stub._prepare_client_task()
../../util/client/common.py:407: in _prepare_client_task
    self._ensure_ref()
../../util/client/common.py:377: in _ensure_ref
    self._ref = ray.worker._put_pickled(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ray.util.client.worker.Worker object at 0x13d96c100>
data = b'\x80\x05\x95\x95\xbf\x00\x00\x00\x00\x00\x00\x8c\x1bray.cloudpickle.cloudpickle\x94\x8c\x14_make_skeleton_class\x94\...0\x00j4\x08\x00\x00j\xef\x07\x00\x00j5\x08\x00\x00j\xf3\x07\x00\x00j6\x08\x00\x00uu\x86\x94\x86R0u}\x94\x86\x94\x86R0.'
client_ref_id = b'\xcc\xa9\xa6\x8aN\x9c\xd2@\x1b\x82U?\xc1\x8c\x89\xfe\xb9'
owner = None

    def _put_pickled(
        self, data, client_ref_id: bytes, owner: Optional[ClientActorHandle] = None
    ):
        req = ray_client_pb2.PutRequest(data=data)
        if client_ref_id is not None:
            req.client_ref_id = client_ref_id
        if owner is not None:
            req.owner_id = owner.actor_ref.id
    
        resp = self.data_client.PutObject(req)
        if not resp.valid:
            try:
>               raise cloudpickle.loads(resp.error)
E               RuntimeError: There is no current event loop in thread 'ray_client_server_0'.

../../util/client/worker.py:510: RuntimeError

 python/ray/serve/tests/test_ray_client.py::test_quickstart_class ⨯50%
```